### ISSUE TITLE
[JENKINS-3217] Somestimes Hudson hangs with "Building on slave..."

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/jenkins#14191]
<p>This happens sometimes: a job gets started on a slave, and after the output of<br/>
"building on slave ..." nothing happens.<br/>
I got a StackTrace output from the slave where this happened, hopefully this helps:</p>

<p>Selected Process: java.exe 4936</p>

<p>Wed Mar 04 15:06:03 2009<br/>
Full thread dump Java HotSpot(TM) 64-Bit Server VM (1.5.0_17-b04 mixed mode):</p>

<p>"StackTrace Remote Thread" prio=6 tid=0x0000000005512610 nid=0x3f88 waiting on<br/>
condition <span class="error">&#91;0x0000000000000000..0x0000000007b9f7f0&#93;</span></p>

<p>"pool-1-thread-178" prio=6 tid=0x0000000005512b70 nid=0xad8 waiting for monitor<br/>
entry <span class="error">&#91;0x0000000005fdf000..0x0000000005fdf8e0&#93;</span><br/>
at hudson.remoting.Channel.send(Channel.java:341)</p>
<ul class="alternate" type="square">
	<li>waiting to lock <0x00000000154141b0> (a hudson.remoting.Channel)<br/>
at hudson.remoting.Request$2.run(Request.java:243)<br/>
at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:417)<br/>
at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:269)<br/>
at java.util.concurrent.FutureTask.run(FutureTask.java:123)<br/>
at<br/>
java.util.concurrent.ThreadPoolExecutor$Worker.runTask(ThreadPoolExecutor.java:650)<br/>
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:675)<br/>
at hudson.remoting.Engine$1$1.run(Engine.java:54)<br/>
at java.lang.Thread.run(Thread.java:595)</li>
</ul>

<p>"pool-1-thread-177" prio=6 tid=0x0000000005513630 nid=0x38a4 waiting for monitor<br/>
entry <span class="error">&#91;0x0000000005ecf000..0x0000000005ecf960&#93;</span><br/>
at hudson.remoting.Channel.send(Channel.java:341)</p>
<ul class="alternate" type="square">
	<li>waiting to lock <0x00000000154141b0> (a hudson.remoting.Channel)<br/>
at hudson.remoting.Request$2.run(Request.java:243)<br/>
at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:417)<br/>
at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:269)<br/>
at java.util.concurrent.FutureTask.run(FutureTask.java:123)<br/>
at<br/>
java.util.concurrent.ThreadPoolExecutor$Worker.runTask(ThreadPoolExecutor.java:650)<br/>
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:675)<br/>
at hudson.remoting.Engine$1$1.run(Engine.java:54)<br/>
at java.lang.Thread.run(Thread.java:595)</li>
</ul>

<p>"pool-1-thread-176" prio=6 tid=0x00000000055128c0 nid=0x5540 runnable<br/>
<span class="error">&#91;0x000000000669e000..0x000000000669f9e0&#93;</span><br/>
at java.net.SocketOutputStream.socketWrite0(Native Method)<br/>
at java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:92)<br/>
at java.net.SocketOutputStream.write(SocketOutputStream.java:136)<br/>
at java.io.BufferedOutputStream.flushBuffer(BufferedOutputStream.java:65)<br/>
at java.io.BufferedOutputStream.flush(BufferedOutputStream.java:123)</p>
<ul class="alternate" type="square">
	<li>locked <0x000000001546a6a0> (a java.io.BufferedOutputStream)<br/>
at<br/>
java.io.ObjectOutputStream$BlockDataOutputStream.flush(ObjectOutputStream.java:1631)<br/>
at java.io.ObjectOutputStream.flush(ObjectOutputStream.java:666)<br/>
at hudson.remoting.Channel.send(Channel.java:348)</li>
	<li>locked <0x00000000154141b0> (a hudson.remoting.Channel)<br/>
at hudson.remoting.ProxyOutputStream.write(ProxyOutputStream.java:112)</li>
	<li>locked <0x0000000016c8ef98> (a hudson.remoting.ProxyOutputStream)<br/>
at hudson.remoting.ProxyOutputStream.write(ProxyOutputStream.java:100)<br/>
at java.io.BufferedOutputStream.flushBuffer(BufferedOutputStream.java:65)<br/>
at java.io.BufferedOutputStream.flush(BufferedOutputStream.java:123)</li>
	<li>locked <0x0000000016c893c8> (a java.io.BufferedOutputStream)<br/>
at java.io.FilterOutputStream.flush(FilterOutputStream.java:123)<br/>
at org.apache.tools.tar.TarBuffer.writeBlock(TarBuffer.java:411)<br/>
at org.apache.tools.tar.TarBuffer.writeRecord(TarBuffer.java:388)<br/>
at org.apache.tools.tar.TarOutputStream.write(TarOutputStream.java:332)<br/>
at hudson.FilePath.writeToTar(FilePath.java:1047)<br/>
at hudson.FilePath.access$500(FilePath.java:145)<br/>
at hudson.FilePath$26.invoke(FilePath.java:982)<br/>
at hudson.FilePath$26.invoke(FilePath.java:979)<br/>
at hudson.FilePath$FileCallableWrapper.call(FilePath.java:1282)<br/>
at hudson.remoting.UserRequest.perform(UserRequest.java:92)<br/>
at hudson.remoting.UserRequest.perform(UserRequest.java:46)<br/>
at hudson.remoting.Request$2.run(Request.java:236)<br/>
at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:417)<br/>
at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:269)<br/>
at java.util.concurrent.FutureTask.run(FutureTask.java:123)<br/>
at<br/>
java.util.concurrent.ThreadPoolExecutor$Worker.runTask(ThreadPoolExecutor.java:650)<br/>
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:675)<br/>
at hudson.remoting.Engine$1$1.run(Engine.java:54)<br/>
at java.lang.Thread.run(Thread.java:595)</li>
</ul>

<p>"Channel reader thread: channel" prio=6 tid=0x00000000004c50b0 nid=0x233c<br/>
runnable <span class="error">&#91;0x000000000539f000..0x000000000539fae0&#93;</span><br/>
at java.net.SocketInputStream.socketRead0(Native Method)<br/>
at java.net.SocketInputStream.read(SocketInputStream.java:129)<br/>
at java.io.BufferedInputStream.fill(BufferedInputStream.java:218)<br/>
at java.io.BufferedInputStream.read(BufferedInputStream.java:235)</p>
<ul class="alternate" type="square">
	<li>locked <0x00000000154160f8> (a java.io.BufferedInputStream)<br/>
at java.io.ObjectInputStream$PeekInputStream.peek(ObjectInputStream.java:2196)<br/>
at java.io.ObjectInputStream$BlockDataInputStream.peek(ObjectInputStream.java:2486)<br/>
at<br/>
java.io.ObjectInputStream$BlockDataInputStream.peekByte(ObjectInputStream.java:2496)<br/>
at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1273)<br/>
at java.io.ObjectInputStream.readObject(ObjectInputStream.java:348)<br/>
at hudson.remoting.Channel$ReaderThread.run(Channel.java:665)</li>
</ul>

<p>"Thread-5" prio=6 tid=0x00000000055120b0 nid=0x2de0 in Object.wait()<br/>
<span class="error">&#91;0x000000000679f000..0x000000000679fb60&#93;</span><br/>
at java.lang.Object.wait(Native Method)</p>
<ul class="alternate" type="square">
	<li>waiting on <0x00000000154141b0> (a hudson.remoting.Channel)<br/>
at java.lang.Object.wait(Object.java:474)<br/>
at hudson.remoting.Channel.join(Channel.java:568)</li>
	<li>locked <0x00000000154141b0> (a hudson.remoting.Channel)<br/>
at hudson.remoting.Engine.run(Engine.java:154)</li>
</ul>

<p>"DestroyJavaVM" prio=6 tid=0x00000000004c4090 nid=0x270 waiting on condition<br/>
<span class="error">&#91;0x0000000000000000..0x000000000023f860&#93;</span></p>

<p>"Wrapper-Connection" daemon prio=10 tid=0x00000000004c5360 nid=0x2018 runnable<br/>
<span class="error">&#91;0x0000000005caf000..0x0000000005caf7e0&#93;</span><br/>
at java.net.SocketInputStream.socketRead0(Native Method)<br/>
at java.net.SocketInputStream.read(SocketInputStream.java:129)<br/>
at java.net.SocketInputStream.read(SocketInputStream.java:182)<br/>
at java.io.DataInputStream.readByte(DataInputStream.java:241)<br/>
at org.tanukisoftware.wrapper.WrapperManager.handleSocket(WrapperManager.java:3737)<br/>
at org.tanukisoftware.wrapper.WrapperManager.run(WrapperManager.java:4084)<br/>
at java.lang.Thread.run(Thread.java:595)</p>

<p>"Wrapper-Control-Event-Monitor" daemon prio=6 tid=0x00000000004c4e00 nid=0x2dc4<br/>
waiting on condition <span class="error">&#91;0x000000000529f000..0x000000000529f8e0&#93;</span><br/>
at java.lang.Thread.sleep(Native Method)<br/>
at org.tanukisoftware.wrapper.WrapperManager$3.run(WrapperManager.java:731)</p>

<p>"Low Memory Detector" daemon prio=6 tid=0x00000000004c4b50 nid=0x1774 runnable<br/>
<span class="error">&#91;0x0000000000000000..0x0000000000000000&#93;</span></p>

<p>"CompilerThread1" daemon prio=10 tid=0x00000000043b0150 nid=0x1b4 waiting on<br/>
condition <span class="error">&#91;0x0000000000000000..0x0000000004b8ebb0&#93;</span></p>

<p>"CompilerThread0" daemon prio=10 tid=0x00000000043afe80 nid=0x29b8 waiting on<br/>
condition <span class="error">&#91;0x0000000000000000..0x0000000004a8ec40&#93;</span></p>

<p>"AdapterThread" daemon prio=10 tid=0x00000000043afbb0 nid=0x2bcc waiting on<br/>
condition <span class="error">&#91;0x0000000000000000..0x0000000000000000&#93;</span></p>

<p>"Signal Dispatcher" daemon prio=10 tid=0x00000000004c48a0 nid=0x1e74 runnable<br/>
<span class="error">&#91;0x0000000000000000..0x0000000000000000&#93;</span></p>

<p>"Finalizer" daemon prio=8 tid=0x00000000004c45f0 nid=0x26a4 in Object.wait()<br/>
<span class="error">&#91;0x000000000478f000..0x000000000478f860&#93;</span><br/>
at java.lang.Object.wait(Native Method)</p>
<ul class="alternate" type="square">
	<li>waiting on <0x000000001542c8e8> (a java.lang.ref.ReferenceQueue$Lock)<br/>
at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:120)</li>
	<li>locked <0x000000001542c8e8> (a java.lang.ref.ReferenceQueue$Lock)<br/>
at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:136)<br/>
at java.lang.ref.Finalizer$FinalizerThread.run(Finalizer.java:159)</li>
</ul>

<p>"Reference Handler" daemon prio=10 tid=0x00000000004c4340 nid=0x18c0 in<br/>
Object.wait() <span class="error">&#91;0x000000000468f000..0x000000000468f8e0&#93;</span><br/>
at java.lang.Object.wait(Native Method)</p>
<ul class="alternate" type="square">
	<li>waiting on <0x0000000015414090> (a java.lang.ref.Reference$Lock)<br/>
at java.lang.Object.wait(Object.java:474)<br/>
at java.lang.ref.Reference$ReferenceHandler.run(Reference.java:116)</li>
	<li>locked <0x0000000015414090> (a java.lang.ref.Reference$Lock)</li>
</ul>

<p>"VM Thread" prio=10 tid=0x00000000004bf1b0 nid=0xa64 runnable</p>

<p>"GC task thread#0 (ParallelGC)" prio=6 tid=0x00000000003e30d0 nid=0x1b78 runnable</p>

<p>"GC task thread#1 (ParallelGC)" prio=6 tid=0x00000000003e3350 nid=0x2e64 runnable</p>

<p>"GC task thread#2 (ParallelGC)" prio=6 tid=0x00000000003e35d0 nid=0x2f08 runnable</p>

<p>"GC task thread#3 (ParallelGC)" prio=6 tid=0x00000000003e3850 nid=0x62c runnable</p>

<p>"VM Periodic Task Thread" prio=10 tid=0x00000000004bf2d0 nid=0x201c waiting on<br/>
condition</p>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/vessel.png" title="vessel's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=vessel">vessel</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-3217" target="_blank">Somestimes Hudson hangs with "Building on slave..."</a></i></summary>
<i><ul>
<li><b>status</b>: Closed
<li><b>priority</b>: Critical
<li><b>component(s)</b>: core
<li><b>resolution</b>: Duplicate
<li><b>resolved</b>: 2009-04-05T01:43:58+00:00
<li><b>votes</b>: 0
<li><b>watchers</b>: 0
<li><b>imported</b>: 2025-11-24
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>This happens sometimes: a job gets started on a slave, and after the output of
"building on slave ..." nothing happens.
I got a StackTrace output from the slave where this happened, hopefully this helps:</p>

<p>Selected Process: java.exe 4936</p>

<p>Wed Mar 04 15:06:03 2009
Full thread dump Java HotSpot(TM) 64-Bit Server VM (1.5.0_17-b04 mixed mode):</p>

<p>"StackTrace Remote Thread" prio=6 tid=0x0000000005512610 nid=0x3f88 waiting on
condition <span class="error">&#91;0x0000000000000000..0x0000000007b9f7f0&#93;</span></p>

<p>"pool-1-thread-178" prio=6 tid=0x0000000005512b70 nid=0xad8 waiting for monitor
entry <span class="error">&#91;0x0000000005fdf000..0x0000000005fdf8e0&#93;</span>
at hudson.remoting.Channel.send(Channel.java:341)</p>
<ul class="alternate" type="square">
	<li>waiting to lock &lt;0x00000000154141b0&gt; (a hudson.remoting.Channel)
at hudson.remoting.Request$2.run(Request.java:243)
at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:417)
at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:269)
at java.util.concurrent.FutureTask.run(FutureTask.java:123)
at
java.util.concurrent.ThreadPoolExecutor$Worker.runTask(ThreadPoolExecutor.java:650)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:675)
at hudson.remoting.Engine$1$1.run(Engine.java:54)
at java.lang.Thread.run(Thread.java:595)</li>
</ul>

<p>"pool-1-thread-177" prio=6 tid=0x0000000005513630 nid=0x38a4 waiting for monitor
entry <span class="error">&#91;0x0000000005ecf000..0x0000000005ecf960&#93;</span>
at hudson.remoting.Channel.send(Channel.java:341)</p>
<ul class="alternate" type="square">
	<li>waiting to lock &lt;0x00000000154141b0&gt; (a hudson.remoting.Channel)
at hudson.remoting.Request$2.run(Request.java:243)
at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:417)
at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:269)
at java.util.concurrent.FutureTask.run(FutureTask.java:123)
at
java.util.concurrent.ThreadPoolExecutor$Worker.runTask(ThreadPoolExecutor.java:650)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:675)
at hudson.remoting.Engine$1$1.run(Engine.java:54)
at java.lang.Thread.run(Thread.java:595)</li>
</ul>

<p>"pool-1-thread-176" prio=6 tid=0x00000000055128c0 nid=0x5540 runnable
<span class="error">&#91;0x000000000669e000..0x000000000669f9e0&#93;</span>
at java.net.SocketOutputStream.socketWrite0(Native Method)
at java.net.SocketOutputStream.socketWrite(SocketOutputStream.java:92)
at java.net.SocketOutputStream.write(SocketOutputStream.java:136)
at java.io.BufferedOutputStream.flushBuffer(BufferedOutputStream.java:65)
at java.io.BufferedOutputStream.flush(BufferedOutputStream.java:123)</p>
<ul class="alternate" type="square">
	<li>locked &lt;0x000000001546a6a0&gt; (a java.io.BufferedOutputStream)
at
java.io.ObjectOutputStream$BlockDataOutputStream.flush(ObjectOutputStream.java:1631)
at java.io.ObjectOutputStream.flush(ObjectOutputStream.java:666)
at hudson.remoting.Channel.send(Channel.java:348)</li>
	<li>locked &lt;0x00000000154141b0&gt; (a hudson.remoting.Channel)
at hudson.remoting.ProxyOutputStream.write(ProxyOutputStream.java:112)</li>
	<li>locked &lt;0x0000000016c8ef98&gt; (a hudson.remoting.ProxyOutputStream)
at hudson.remoting.ProxyOutputStream.write(ProxyOutputStream.java:100)
at java.io.BufferedOutputStream.flushBuffer(BufferedOutputStream.java:65)
at java.io.BufferedOutputStream.flush(BufferedOutputStream.java:123)</li>
	<li>locked &lt;0x0000000016c893c8&gt; (a java.io.BufferedOutputStream)
at java.io.FilterOutputStream.flush(FilterOutputStream.java:123)
at org.apache.tools.tar.TarBuffer.writeBlock(TarBuffer.java:411)
at org.apache.tools.tar.TarBuffer.writeRecord(TarBuffer.java:388)
at org.apache.tools.tar.TarOutputStream.write(TarOutputStream.java:332)
at hudson.FilePath.writeToTar(FilePath.java:1047)
at hudson.FilePath.access$500(FilePath.java:145)
at hudson.FilePath$26.invoke(FilePath.java:982)
at hudson.FilePath$26.invoke(FilePath.java:979)
at hudson.FilePath$FileCallableWrapper.call(FilePath.java:1282)
at hudson.remoting.UserRequest.perform(UserRequest.java:92)
at hudson.remoting.UserRequest.perform(UserRequest.java:46)
at hudson.remoting.Request$2.run(Request.java:236)
at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:417)
at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:269)
at java.util.concurrent.FutureTask.run(FutureTask.java:123)
at
java.util.concurrent.ThreadPoolExecutor$Worker.runTask(ThreadPoolExecutor.java:650)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:675)
at hudson.remoting.Engine$1$1.run(Engine.java:54)
at java.lang.Thread.run(Thread.java:595)</li>
</ul>

<p>"Channel reader thread: channel" prio=6 tid=0x00000000004c50b0 nid=0x233c
runnable <span class="error">&#91;0x000000000539f000..0x000000000539fae0&#93;</span>
at java.net.SocketInputStream.socketRead0(Native Method)
at java.net.SocketInputStream.read(SocketInputStream.java:129)
at java.io.BufferedInputStream.fill(BufferedInputStream.java:218)
at java.io.BufferedInputStream.read(BufferedInputStream.java:235)</p>
<ul class="alternate" type="square">
	<li>locked &lt;0x00000000154160f8&gt; (a java.io.BufferedInputStream)
at java.io.ObjectInputStream$PeekInputStream.peek(ObjectInputStream.java:2196)
at java.io.ObjectInputStream$BlockDataInputStream.peek(ObjectInputStream.java:2486)
at
java.io.ObjectInputStream$BlockDataInputStream.peekByte(ObjectInputStream.java:2496)
at java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1273)
at java.io.ObjectInputStream.readObject(ObjectInputStream.java:348)
at hudson.remoting.Channel$ReaderThread.run(Channel.java:665)</li>
</ul>

<p>"Thread-5" prio=6 tid=0x00000000055120b0 nid=0x2de0 in Object.wait()
<span class="error">&#91;0x000000000679f000..0x000000000679fb60&#93;</span>
at java.lang.Object.wait(Native Method)</p>
<ul class="alternate" type="square">
	<li>waiting on &lt;0x00000000154141b0&gt; (a hudson.remoting.Channel)
at java.lang.Object.wait(Object.java:474)
at hudson.remoting.Channel.join(Channel.java:568)</li>
	<li>locked &lt;0x00000000154141b0&gt; (a hudson.remoting.Channel)
at hudson.remoting.Engine.run(Engine.java:154)</li>
</ul>

<p>"DestroyJavaVM" prio=6 tid=0x00000000004c4090 nid=0x270 waiting on condition
<span class="error">&#91;0x0000000000000000..0x000000000023f860&#93;</span></p>

<p>"Wrapper-Connection" daemon prio=10 tid=0x00000000004c5360 nid=0x2018 runnable
<span class="error">&#91;0x0000000005caf000..0x0000000005caf7e0&#93;</span>
at java.net.SocketInputStream.socketRead0(Native Method)
at java.net.SocketInputStream.read(SocketInputStream.java:129)
at java.net.SocketInputStream.read(SocketInputStream.java:182)
at java.io.DataInputStream.readByte(DataInputStream.java:241)
at org.tanukisoftware.wrapper.WrapperManager.handleSocket(WrapperManager.java:3737)
at org.tanukisoftware.wrapper.WrapperManager.run(WrapperManager.java:4084)
at java.lang.Thread.run(Thread.java:595)</p>

<p>"Wrapper-Control-Event-Monitor" daemon prio=6 tid=0x00000000004c4e00 nid=0x2dc4
waiting on condition <span class="error">&#91;0x000000000529f000..0x000000000529f8e0&#93;</span>
at java.lang.Thread.sleep(Native Method)
at org.tanukisoftware.wrapper.WrapperManager$3.run(WrapperManager.java:731)</p>

<p>"Low Memory Detector" daemon prio=6 tid=0x00000000004c4b50 nid=0x1774 runnable
<span class="error">&#91;0x0000000000000000..0x0000000000000000&#93;</span></p>

<p>"CompilerThread1" daemon prio=10 tid=0x00000000043b0150 nid=0x1b4 waiting on
condition <span class="error">&#91;0x0000000000000000..0x0000000004b8ebb0&#93;</span></p>

<p>"CompilerThread0" daemon prio=10 tid=0x00000000043afe80 nid=0x29b8 waiting on
condition <span class="error">&#91;0x0000000000000000..0x0000000004a8ec40&#93;</span></p>

<p>"AdapterThread" daemon prio=10 tid=0x00000000043afbb0 nid=0x2bcc waiting on
condition <span class="error">&#91;0x0000000000000000..0x0000000000000000&#93;</span></p>

<p>"Signal Dispatcher" daemon prio=10 tid=0x00000000004c48a0 nid=0x1e74 runnable
<span class="error">&#91;0x0000000000000000..0x0000000000000000&#93;</span></p>

<p>"Finalizer" daemon prio=8 tid=0x00000000004c45f0 nid=0x26a4 in Object.wait()
<span class="error">&#91;0x000000000478f000..0x000000000478f860&#93;</span>
at java.lang.Object.wait(Native Method)</p>
<ul class="alternate" type="square">
	<li>waiting on &lt;0x000000001542c8e8&gt; (a java.lang.ref.ReferenceQueue$Lock)
at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:120)</li>
	<li>locked &lt;0x000000001542c8e8&gt; (a java.lang.ref.ReferenceQueue$Lock)
at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:136)
at java.lang.ref.Finalizer$FinalizerThread.run(Finalizer.java:159)</li>
</ul>

<p>"Reference Handler" daemon prio=10 tid=0x00000000004c4340 nid=0x18c0 in
Object.wait() <span class="error">&#91;0x000000000468f000..0x000000000468f8e0&#93;</span>
at java.lang.Object.wait(Native Method)</p>
<ul class="alternate" type="square">
	<li>waiting on &lt;0x0000000015414090&gt; (a java.lang.ref.Reference$Lock)
at java.lang.Object.wait(Object.java:474)
at java.lang.ref.Reference$ReferenceHandler.run(Reference.java:116)</li>
	<li>locked &lt;0x0000000015414090&gt; (a java.lang.ref.Reference$Lock)</li>
</ul>

<p>"VM Thread" prio=10 tid=0x00000000004bf1b0 nid=0xa64 runnable</p>

<p>"GC task thread#0 (ParallelGC)" prio=6 tid=0x00000000003e30d0 nid=0x1b78 runnable</p>

<p>"GC task thread#1 (ParallelGC)" prio=6 tid=0x00000000003e3350 nid=0x2e64 runnable</p>

<p>"GC task thread#2 (ParallelGC)" prio=6 tid=0x00000000003e35d0 nid=0x2f08 runnable</p>

<p>"GC task thread#3 (ParallelGC)" prio=6 tid=0x00000000003e3850 nid=0x62c runnable</p>

<p>"VM Periodic Task Thread" prio=10 tid=0x00000000004bf2d0 nid=0x201c waiting on
condition</p></pre>
</details>
</details>
<ul><li><i>environment</i>: <code>Platform: All, OS: All</code></li></ul>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>mdonohue:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-3217?focusedId=129389&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-129389">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Issue 2397 covers this</p>

<ul>
	<li>
	<ul>
		<li>
		<ul>
			<li>This issue has been marked as a duplicate of 2397 ***</li>
		</ul>
		</li>
	</ul>
	</li>
</ul>

</pre>
</details>

</ul>
</details>

<p>Issue 2397 covers this</p>

<ul>
	<li>
	<ul>
		<li>
		<ul>
			<li>This issue has been marked as a duplicate of 2397 ***</li>
		</ul>
		</li>
	</ul>
	</li>
</ul>

**Comment by jenkins-infra-bot** [comment]
<i>[Original `duplicates` from Jira: <a href="https://github.com/jenkinsci/jenkins/issues?q=is%3Aissue%20%22jira_issue_key%3DJENKINS-2397%22">JENKINS-2397</a>]</i>
